### ISSUE TITLE
Chat box text remains after cancelling the message with esc

### ISSUE THREAD

**Issue by cu-kai** [Unvanquished/Unvanquished#2018]
When typing text into any messagemode and pressing esc to cancel instead of enter to send, when any messagemode is used again, the text persists. This has only happened since 0.53.0's release.

Screenshot for the avoidance of doubt:
![image](https://user-images.githubusercontent.com/13281185/182442678-01631c49-854f-4c7d-9b46-1e2ef3453417.png)

**Comment by DolceTriade** [comment]
Can confirm. I have repro'd this as well. Fix seems trivial tho.

**Comment by DolceTriade** [comment]
Basically we must be missing an edgecase for clearing the text buffer when the chat element gets hidden.

**Comment by DolceTriade** [comment]
Should probably add a blue event listener here and call `text.clear()` and `UpdateText()`

https://github.com/Unvanquished/Unvanquished/blob/master/src/cgame/rocket/rocketChatField.h#L77

**Comment by illwieckz** [comment]
If it isn't intentional it's a bug, but I can see some benefit in such mechanism. I noticed I can escape from a public chat and get my already written sentence back to post it as team chat (or admin chat). The main issue is that I don't have a way to easily clear what's written. I can imagine such mechanism as a feature where someone hits Ctrl+C to clear, and some GUI around with tabs to switch between public/team/admin chat while keeping the sentence before sending it, and some clear/send buttons.

But better fix bugs that are bugs first. :D

Also if the retention of the message was not intentional and this is a regression, I'm curious about knowing the root cause of this retention and how it slipped in.

**Comment by slipher** [comment]
Almost every chat app keeps your "draft" message if you switch to another conversation, or (on mobile devices) close the app. So I don't think erasing the message is desirable. You could get harassed by a dretch while writing it :)

**Comment by DolceTriade** [comment]
This was the behavior in tremulous, but I dont' remember the behavior in librocket

**Comment by illwieckz** [comment]
I see the point in keeping the message, bug if we want to keep it, we need a way to clear it.
Either we never keep it, either we keep it and add a way to clear it, like Ctrl+C (already works in console).

**Comment by slipher** [comment]
If we made it a normal text field then we could use Ctrl+A, backspace/delete. I wonder why it was written as a completely custom element in the first place. (There is the need for color support, but the player name editor has that and still takes advantage of the built-in text field.)

**Comment by DolceTriade** [comment]
Every game I've played will clear the message however. I'm happy with a compromise of allowing people to clear the messages with ctrl-c or whatever, but my preference is that esc clears the message.

Also, even if you swap between team chat and regular chat, the messages remains. The reason we wrote this custom element was to preview colors as we type rather than the current rocket chat thing that doesn't show the colors as you type.

**Comment by illwieckz** [comment]
> Also, even if you swap between team chat and regular chat, the messages remains.

I see this as something that may be wanted. With previous behaviour sometime I started to type then I noticed the message would be public, so I had to scrap it and start it entirely from scratch in the team chat instead, losing precious time. And this was even more precious time being lost because in the middle of the game a team chat is usually strategic, not just chatting with others about random things.

All in all, the current situation without any way to clear the field is really annoying. Not being able to clear a now-obsolete non-sent-yet message feels to me much more annoying to me than the need to restart a message from scratch because it was not targeted to the right people.

**Comment by necessarily-equal** [comment]
I'm having trouble telling if it's closed or not, but I don't think anything has to be done before the 0.54.

**Comment by cu-kai** [comment]
I would not consider this closed as the text still remains after closing the chatbox. The CTRL-C is a workaround imo.

**Comment by DolceTriade** [comment]
This should either be closed as "Working as intended" or we should fix it. I did opt to fix it, but I received feedback that this feature was otherwise desired. 🤷

**Comment by necessarily-equal** [comment]
I like being able to continue my mesages if I press escape really quick because I'd get eaten otherwise :eyes: :alien:

**Comment by DolceTriade** [comment]
as for me, I prefer esc canceling the message but the ctrl-c thing is good enough for me.

**Comment by cu-kai** [comment]
I'm sort of used to this behaviour now and don't mind it so much after all these months. Perhaps it should be made optional and also obvious that CTRL+C is required to "clear" the chat box.

**Comment by DolceTriade** [comment]
is fine for me too.

**Comment by illwieckz** [comment]
Yes, it would be cool to add a text saying ”Ctrl+C to clear”, but that can be another issue.

**Comment by illwieckz** [comment]
Now that this is opened:

- https://github.com/Unvanquished/Unvanquished/issues/2477

I guess we can close this one.

**Comment by illwieckz** [comment]
@cu-kai wants to keep it open,

This opened issue is now tracking the ability to enable or disable chat persistence in options.

**Comment by illwieckz** [comment]
I renamed the issue to be closer to the current need.

**Comment by bmorel** [comment]
yes, this clearly should not be closed as it's a clear regression compared to what we had before and what people from any other game I've played to (and there's a lot) will expect.

**Comment by VReaperV** [comment]
Completed by #3386.
